### DISCUSSION TITLE
Changes to service granularity over time?

### DISCUSSION THREAD

**Discussion by famkedriessen** [apolloconfig/apollo#4672] [Q&A]
Hi all! I am doing a research in which I analyze git histories of projects working on microservice-based systems, to see whether such a model can point out services that should be split or merged, based on dependencies in the git logs (if service A is 90% of the time updated on the same day as service B, this might be an indication for tight-coupling). To validate my model I am looking for open-source cases in which either some functionality was extracted into a new service, or in which architectural decay has taken place (a service grew bigger and bigger over time while becoming less and less cohesive). Would someone be able to point out such "fluctuations in granularity" in Apollo to me? Thank you in advance!

**Comment by nobodyiam** [comment]
I'm not aware of the architectural decay in apollo, but they could exist and you may start from the [biz/service](https://github.com/apolloconfig/apollo/tree/master/apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service) and [portal/service](https://github.com/apolloconfig/apollo/tree/master/apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service) packages.

**Reply by famkedriessen** [reply]
Thank you, I will look into those. So next to potential decay, there has never been done any refactoring in which functionality is moved from one service to another/a new one?

**Reply by nobodyiam** [reply]
not recently.
